### DISCUSSION TITLE
Embedded WYSIWYG page/content editor module for static content

### DISCUSSION THREAD

**Discussion by kudryavtsevda** [OrchardCMS/OrchardCore#11215] [Q&A] [answered]
I am curious if there is a way to create a module which allows "content manager" to customize page layout using wysiwyg editor like  [grapes js](https://grapesjs.com/) at least only static content like background image/css/html tags?

**Comment by Skrypt** [comment, marked as answer]
You could implement a new custom editor for an `HtmlBody` that uses this Grape.js script.

**Reply by kudryavtsevda** [reply]
> You could implement a new custom editor for an `HtmlBody` that uses this Grape.js script.

does it mean that a new content type called "grape js content" should be created which consists of one part "GrapeJS Content" with only one field inside and that field should be "HtmlBody"? I am a novice in orchard and just trying to understand the design of the solution.

**Reply by Skrypt** [reply]
No, not exactly.
Let's say you have a Content Type named "Acme". Then you add an HtmlBody part to your content type. A content part is not a Field, these are 2 different concepts. Then, if you go to configure your HtmlBody part you will see that you can decide which editor you prefer ... Monaco, Trumbowyg ...

You can create a custom module that will add new Editor options in that list.

Though one thing that I'm not sure about is the way that the grape.js editor will display when you want to create your new content items of that Content Type. The editor seems to always display in Full page. Maybe there are options to make it contained inside a smaller layout element.

**Reply by kudryavtsevda** [reply]
I thought it could be kind of popup full size window which contains grapes.js editor inside. And when a page layout design is done then content manager can apply changes by pressing "save" button which runs uploading images and css as media on filestorage (or keep it inside as base64 for image) and html content will be saved as html body content. 
thank you for your advice. if you don't mind I will come back when I have something to discuss in more details

**Comment by hishamco** [comment]
I implemented such editor, hope to share it in the upcoming weeks on Orchard Core Contrib (OCC)

**Reply by kudryavtsevda** [reply]
I am eager to see it!
